### ISSUE TITLE
Assertion failed in libnormaliz/project_and_lift.cpp:588 with redundant input sections for normaliz 3.9.4

### ISSUE THREAD

**Issue by benlorenz** [Normaliz/Normaliz#400]
Normaliz 3.9.4 crashes with a failed assertion on the following input:
```
amb_space 9
cone 1
2 1 1 0 0 0 0 0 0
inequalities 9
1 0 0 0 0 0 0 0 0
0 1 0 0 0 0 0 0 0
0 0 1 0 0 0 0 0 0
0 0 0 1 0 0 0 0 0
0 0 0 0 1 0 0 0 0
0 0 0 0 0 1 0 0 0
0 0 0 0 0 0 1 0 0
0 0 0 0 0 0 0 1 0
0 0 0 0 0 0 0 0 1
equations 4
-1 1 1 1 1 1 1 1 1
1 -1 -1 -1 -1 1 1 1 1
1 -1 -1 1 1 1 1 -1 -1
0 1 -1 1 -1 1 -1 -1 1
grading
unit_vector 1
NoGradingDenom
Deg1Elements
```

Running normaliz 3.9.4 on this file triggers the following assertion:
> normaliz: libnormaliz/project_and_lift.cpp:588: std::vector<long unsigned int> libnormaliz::ProjectAndLift<IntegerPL, IntegerRet>::order_supps(const libnormaliz::Matrix<Integer>&) [with IntegerPL = long long int; IntegerRet = long long int]: Assertion `Supps.nr_of_rows() > 0' failed.
> Aborted

Removing either the `cone` section or the `inequalities` + `equations` sections avoids the problem.

Normaliz 3.9.3 also works fine on this file and writes (correctly):
> 0 lattice points in polytope (Hilbert basis elements of degree 1)

**Comment by w-bruns** [comment]
Problem solved. Patch will come tomorrow.

**Comment by w-bruns** [comment]
As a workaround you can use the option `NoSparseProjection` in 3.9.4.

As a temporary fix for the code change line 6293 of `cone.cpp` to

   ```
 if(positive_and_bounded && !ToCompute.test(ConeProperty::NoCoarseProjection)
                && !isComputed(ConeProperty::Generators)){
```
The problem arises since the generator of the cone satisfies the constraints, and these are therefore removed. But before they had been used already to set `positive_and_bounded`. 

Thanks for the test!

**Comment by w-bruns** [comment]
Fixed in 3.10.0.
